src/hooks/useDebounce.tsx

```diff
@@ -0,0 +1,14 @@
+import { useState, useEffect } from "react";
+
+export const useDebounce = <X, Y extends number>(value: X, delay: Y) => {
```

Comment:
You need generic for value, because you want to indicate that

When value is `string` then returned value is `string`;
When value is `number` then returned value is `number`;
When value is `{ someKey: string }` then returned value is `{ someKey: string }`;

But you don't need generic for `delay`, because it will always be number. So it can just be `export const useDebounce = <X>(value: X, delay: number): X => {`

Another small thing, you can give meaningful name for generic parameter, so for example `export const useDebounce = <Value>(value: Value, delay: number): Value => {`